sys/include/ut_process.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2021 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup sys_ut_process URI template processor
+ * @ingroup  sys
+ * @brief   URI template processor
+ *
+ * A parser and processor for URI templates up to level 3 expression types
+ * according to [RFC 6570](https://tools.ietf.org/html/rfc6570).
+ *
+ * @see [RFC 6570](https://tools.ietf.org/html/rfc6570)
+ *
+ * @{
+ *
+ * @file
+ * @brief   URI template processor definitions
+ *
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+#ifndef UT_PROCESS_H
+#define UT_PROCESS_H
+
+#include <stddef.h>
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Name-value-pair of a variable for URI template extension
+ */
+typedef struct {
+    const char *name;       /**< name of the variable */
+    const char *value;      /**< value for the variable */
+} ut_process_var_t;
+
+/**
+ * @brief   Extends a URI template by a given value set
```

Comment:
What does 'Extend' mean? Since this is the only function this module provides, I would expect it also does some template matching?

Reply:
Apparently I was mistaking. The word 'extend' only appears once in the RFC and not in the context of setting the variables... Let me check again.

Reply:
Ahhhh the word I meant to use here was "expand". Will fix!

Reply:
Fixed! Luckily, `strlen("extend") == strlen("expand")`, so a simple search and replace and no formatting changes were required due to this flub-up. :'-).

Reply:
> It means that a variable is "set" or alternatively, the variable is removed altogether, if it is not provided with the value set.

Why not also add this to the API doc?

Reply:
Do we explain what TCP means or how routing works as well? The word is used within the first sentence of the abstract of the RFC:

>    A URI Template is a compact sequence of characters for describing a
>    range of Uniform Resource Identifiers through variable expansion.

As such, I don't think it needs explaining.

Reply:
>    expansion:  The string result obtained from a template expression
>      after processing it according to its expression type, list of
>      variable names, and value modifiers, as defined in Section 3.

Reply:
As a developer, I like it when an API documentation saves me time. And unlike TCP or routing, it's not common knowledge what a "URI template parser" does.

Anyway, feel free to squash.

Reply:
> As a developer, I like it when an API documentation saves me time. And unlike TCP or routing, it's not common knowledge what a "URI template parser" does.

Come on. If as _a developer_ you don't know, what a variable expansion is, maybe you should revisit programming 101 ;-P.